### PR TITLE
🤬 Add various UI error messages

### PR DESCRIPTION
**Pull request by henryboldi**
This PR includes error messages for,
- [x] Password matches + minimum length (8) in the create password screen.
- [x] Incorrect password in the unlock screen
- [x] Insufficient balance for asset selectors

Plus a spinner for,
- [x] Create password screen

error

error 2

error 3

error 4

### CODE REVIEW

File: ui/components/Keyring/KeyringSetPassword.tsx [hunk 1]
```diff
@@ -21,11 +22,31 @@ export default function KeyringSetPassword(): ReactElement {
     }
   }, [history, areKeyringsUnlocked])
 
+  const validatePassword = (): boolean => {
+    if (password.length < 8) {
```
Comment (greg-nagy) [inline ui/components/Keyring/KeyringSetPassword.tsx:26]: 🤔 If we want to have stronger password requirements?

File: ui/components/Shared/SharedAssetInput.tsx [hunk 2]
```diff
@@ -170,6 +173,19 @@ export default function SharedAssetInput(
     [onAssetSelect]
   )
 
+  const selectedTokenAssetAmount = assetAmounts?.filter((token) => {
+    return token.asset.symbol === selectedToken.symbol
+  })[0]
+
+  const getErrorMessage = (givenAmount: string): string | undefined => {
+    return (!isTypeDestination &&
+      selectedTokenAssetAmount &&
```
Comment (greg-nagy) [inline ui/components/Shared/SharedAssetInput.tsx:182]: It nice have trust, but it's even better when we make sure everything is as expected 🧡

File: ui/components/Shared/SharedAssetInput.tsx [hunk 3]
```diff
@@ -179,10 +195,12 @@ export default function SharedAssetInput(
           setOpenAssetMenu(false)
         }}
       >
-        <SelectTokenMenuContent
-          assets={assets}
-          setSelectedTokenAndClose={setSelectedTokenAndClose}
-        />
+        {assets && (
+          <SelectTokenMenuContent
+            assets={assets}
```
Comment (greg-nagy) [inline ui/components/Shared/SharedAssetInput.tsx:200]: Could it cause any issue here if the assets were the default `[{ symbol: "ETH", name: "Example Asset" }]`?

**Review by greg-nagy — DISMISSED** [review]
This is a nice PR, thanks for this! Let's merge this bad boy :)

**Review by greg-nagy — APPROVED** [review]
let's merge this after conflicts have been resolved

File: ui/components/Shared/SharedAssetInput.tsx (hunk 2 above)
Comment (Shadowfiend) [reply ui/components/Shared/SharedAssetInput.tsx:182]: Yaaasss. Only notes I'll add:
- avoid ternary operators outside of places they're critical.
- typeof === "undefined" is the explicit way we're trying to do undefined checks.

I'm having to flag the second one too much so if I can't find a linter rule for it we might drop it.

**Pull request merged**